src/OrchardCore.Modules/OrchardCore.ContentFields/Indexing/ContentPickerFieldIndexProvider.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using OrchardCore.ContentFields.Fields;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Metadata;
+using OrchardCore.Data;
+using YesSql.Indexes;
+
+namespace OrchardCore.ContentFields.Indexing
+{
+    // Remark: 
+
+    public class ContentPickerFieldIndex : ContentFieldIndex
```

Comment:
Look at the TaxonomiesField, it is a list of ContentItemId instead so you can filter them. You might no be able to inherit from the same base class.

Reply:
Ok you want one row per ContentItemId. Though my solution would still work if using a LIKE '%ContentItemIdString%'